package-system/vulkan-validationlayers/build_config.json

```diff
@@ -0,0 +1,59 @@
+{
+    "git_url":"https://github.com/KhronosGroup/Vulkan-ValidationLayers.git",
+    "git_tag":"v1.2.198",
+    "package_name":"vulkan-validationlayers",
+    "package_version":"1.2.198",
+    "package_url":"https://vulkan.lunarg.com/",
+    "package_license":"Apache-2.0",
+    "package_license_file":"LICENSE.txt",
+    "cmake_find_source":"Findvulkan-validationlayers.cmake",
+    "cmake_find_target":"Findvulkan-validationlayers.cmake",
+    "cmake_src_subfolder": "vulkan-validation",
+    "cmake_build_args" : [ "-j", "8"],
+    "cmake_generate_args_common": [],
+    "cmake_build_args_common": [
```

Comment:
These fields are unused if a custom build command is used (in which case INSTEAD of calling cmake, it just invokes the cmd file instead)

Reply:
If I don't have these entries, errors such as:
Required key 'cmake_generate_args_common' not found in build config

will show up. What would be the correct way of handling this?

Reply:
That shouldn't be the case. For example, the build_config.json for sqlite does not have this key at all because it only uses custom build cmds

Reply:
I tried running the script on SQLite and got similar error. Maybe I am running the wrong script or wrong arguments?

PS E:\3p\3p-package-source> C:\Users\tonypeng\AppData\Local\Programs\Python\Python38\python.exe ..\3p-package-scripts\o3de_package_scripts\build_package.py --search_path E:\3p\3p-package-source SQLite-3.37.2-rev1-windows
Output folder for packages is 'E:\3p\3p-package-source\packages' - Override with --output_folder
Loading windows package lists from E:\3p\3p-package-source...
Loading and merging E:\3p\3p-package-source\package_build_list_host_windows.json ...
Calling build script: "E:\3p\3p-package-source\Scripts\extras\pull_and_build_from_git.py ../../package-system/sqlite --platform-name Windows --package-root ../../package-system --clean"...
Command: cmake --version
Detected CMake: cmake version 3.20.0
Required key 'cmake_src_subfolder' not found in build config
Package SQLite-3.37.2-rev1-windows failed to build from source.

Reply:
Sounds like a bug, maybe I introduced one.  Its not supposed to be a required key.  So its a bug in the script to require it.

Reply:
I think we should fix pull_and_build_from_git.py in a separate PR to have a better track of its history.

Reply:
PR for that fix is here: https://github.com/o3de/3p-package-source/pull/82

Reply:
Took out the fixed optional arguments.